src/js/camresolutionstest.js

```diff
@@ -36,148 +20,109 @@ var resolutions = [[160, 120, false],
+      this.test.done();
+    }.bind(this));
   },
 
-  collectAndAnalyzeStats_: function(stream, selectedResolution) {
+  collectAndAnalyzeStats_: function(stream, resolution) {
     var tracks = stream.getVideoTracks();
     if (tracks.length < 1) {
-      reportError('No video track in returned stream.');
+      this.test.reportError('No video track in returned stream.');
       this.finishTestOrRetrigger_();
       return;
     }
 
     var videoTrack = tracks[0];
     // Register events.
-    videoTrack.onended = function() {
-      reportError('Video track ended, camera stopped working');
-    };
-    videoTrack.onmute = function() {
-      reportError('Your camera reported itself as muted.');
+    videoTrack.addEventListener('ended', function() {
+      this.test.reportError('Video track ended, camera stopped working');
+    });
+    videoTrack.addEventListener('mute', function() {
+      this.test.reportError('Your camera reported itself as muted.');
       // MediaStreamTrack.muted property is not wired up in Chrome yet, checking
       // isMuted local state.
       this.isMuted = true;
-    };
-    videoTrack.onunmute = function() {
+    });
+    videoTrack.addEventListener('unmute', function() {
       this.isMuted = false;
-    };
+    });
 
     var video = document.createElement('video');
     video.setAttribute('autoplay', '');
     video.setAttribute('muted', '');
     window.videoElement = video;
     window.stream = stream;
-    video.width = selectedResolution[0];
-    video.height = selectedResolution[1];
+    video.width = resolution[0];
+    video.height = resolution[1];
     attachMediaStream(video, stream);
     var frameChecker = new VideoFrameChecker(video);
     var call = new Call();
     call.pc1.addStream(stream);
     call.establishConnection();
     call.gatherStats(call.pc1,
-                     this.onCallEnded_.bind(this, selectedResolution, video,
-                                            this.stream, frameChecker),
+                     this.onCallEnded_.bind(this, resolution, video,
+                                            stream, frameChecker),
                      100);
 
     setTimeoutWithProgressBar(call.close.bind(call), 8000);
   },
 
-  onCallEnded_: function(selectedResolution, videoElement, stream, frameChecker,
+  onCallEnded_: function(resolution, videoElement, stream, frameChecker,
                          stats, statsTime) {
-    this.analyzeStats_(selectedResolution, videoElement, stream, frameChecker,
+    this.analyzeStats_(resolution, videoElement, stream, frameChecker,
                        stats, statsTime);
 
-    this.stream.getVideoTracks()[0].onended = null;
-    this.stream.getVideoTracks()[0].onmute = null;
-    this.stream.getVideoTracks()[0].onunmute = null;
-    this.stream.getVideoTracks()[0].stop();
-
     frameChecker.stop();
 
-    this.finishTestOrRetrigger_();
+    stream.getTracks().forEach(function(track) {
+      track.onended = null;
```

Comment:
We never ser onended onmute or onunmute attributes. Why unset them?
I guess you mean to removeEventListener. But are those actually needed? won't those references go away once we stop the track?